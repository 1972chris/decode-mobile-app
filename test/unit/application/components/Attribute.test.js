import React from 'react';
import renderer from 'react-test-renderer';
import AttributeComponent from '../../../../application/components/Attribute/Attribute';

it('renders Attribute on verfied mode component', () => {
  const rendered = renderer.create(<AttributeComponent isVerified />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it('renders Attribute on non-verfied mode component', () => {
  const rendered = renderer.create(<AttributeComponent isVerified={false} />).toJSON();
  expect(rendered).toMatchSnapshot();
});
