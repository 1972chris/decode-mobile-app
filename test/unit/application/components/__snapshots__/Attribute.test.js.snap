// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Attribute on non-verfied mode component 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#FFF",
      "padding": 16,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "flex": 1,
        "fontSize": 16,
        "fontWeight": "500",
      }
    }
  >
    Verified Atlantis Resident*
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.54)",
        "marginVertical": 4,
      }
    }
  >
    To get this attribute you will be directed to the Atlantis Council website
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "rgb(0,163,158)",
        "borderRadius": 2,
        "elevation": 2,
        "height": 36,
        "marginBottom": 18,
        "marginHorizontal": 16,
        "marginTop": 10,
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
        "shadowColor": "rgba(0, 0, 0, 0.54)",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 2,
        "width": 100,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "color": "#FFF",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      REQUEST
    </Text>
  </View>
</View>
`;

exports[`renders Attribute on verfied mode component 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#FFF",
      "display": "flex",
      "flexDirection": "column",
      "marginVertical": 20,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "flex": 1,
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
    >
      Your residency status (required)
    </Text>
    <Switch
      onValueChange={[Function]}
      value={true}
    />
  </View>
</View>
`;
